src/Aspire.Hosting/ApplicationModel/Resource.cs

```diff
@@ -8,20 +8,76 @@ namespace Aspire.Hosting.ApplicationModel;
 /// <summary>
 /// Represents an abstract resource that can be used by an application, that implements <see cref="IResource"/>.
 /// </summary>
-/// <param name="name">The name of the resource.</param>
 [DebuggerDisplay("{DebuggerToString(),nq}")]
-public abstract class Resource(string name) : IResource
+public abstract class Resource : IResource
 {
     /// <summary>
     /// Gets the name of the resource.
     /// </summary>
-    public string Name { get; } = name;
+    public string Name { get; }
     
     /// <summary>
     /// Gets the annotations associated with the resource.
     /// </summary>
     public ResourceMetadataCollection Annotations { get; } = new();
 
+    /// <param name="name">The name of the resource.</param>
+    public Resource(string name)
+    {
+        ValidateName(name);
+
+        Name = name;
+    }
+
+    /// <summary>
+    /// Validate that a resource name is valid.
+    /// - Must start with an ASCII letter.
+    /// - Must contain only ASCII letters, digits, and hyphens.
+    /// - Must not end with a hyphen.
+    /// - Must not contain consecutive hyphens.
+    /// - Must be between 1 and 64 characters long.
+    /// </summary>
+    internal static void ValidateName(string name)
+    {
+        ArgumentException.ThrowIfNullOrEmpty(name, nameof(name));
+
+        if (name.Length > 64)
+        {
+            throw new ArgumentException($"Resource name '{name}' is invalid. Name must be between 1 and 64 characters long.", nameof(name));
```

Comment:
Random question, have we not globalized strings in `Aspire.Hosting` (or elsewhere in this repo) yet?

Reply:
They're in the dashboard.